src/Management/src/EndpointCore/Health/HealthEndpointMiddleware.cs

```diff
@@ -59,7 +60,11 @@ protected internal async Task HandleHealthRequestAsync(HttpContext context)
         protected internal string DoRequest(HttpContext context)
         {
             var result = _endpoint.Invoke(new CoreSecurityContext(context));
-            context.Response.StatusCode = ((HealthEndpoint)_endpoint).GetStatusCode(result);
+            if (((HealthEndpointOptions)_mgmtOptions.FirstOrDefault().EndpointOptions.FirstOrDefault(o => o is HealthEndpointOptions)).HttpStatusFromHealth)
```

Comment:
This does not look correct to me.  I am thinking either filter by type for ActuatorManagementOptions since you don't care about Cloud Foundry path (on `/cloudfoundryapplication`)  or find the matching management options like [here]( https://github.com/SteeltoeOSS/Steeltoe/blob/f2b82898a0f29a3a69d9128b09a33e9299ea4a1b/src/Management/src/EndpointBase/EndpointExtensions.cs#L73). Depending on the setup, the two collections of  EndpointOptions on IManagementOptions can be different and have weird behavior.

Reply:
The only property we care about here is this new `HttpStatusFromHealth`... Could `HealthEndpointOptions` have different values for this property between ActuatorManagementOptions entries and/or is it even possible for these options to be present in only one of the two places?

FWIW, this work was specifically requested for Cloud Foundry use cases